solana-programs/Dockerfile.dev

```diff
@@ -1,58 +1,114 @@
-FROM --platform=linux/amd64 rust:1.64.0 AS bpf-builder
+FROM --platform=linux/amd64 rust:1.60.0 AS bpf-builder
 
 WORKDIR /usr/src/app
 
 ENV PATH="/root/solana-release/bin:${PATH}"
-RUN curl -fsSL https://deb.nodesource.com/setup_14.x | bash - && \
-    apt-get install -y build-essential libudev-dev libhidapi-dev pkg-config libssl-dev git nodejs && \
-    npm install -g yarn@1.22.19 @project-serum/anchor-cli@0.24.1 && \
-    curl -SfL https://github.com/solana-labs/solana/releases/download/v1.10.11/solana-release-x86_64-unknown-linux-gnu.tar.bz2 | tar jxf - -C $HOME
+RUN apt-get update && \
+    apt-get install -y build-essential libudev-dev libhidapi-dev pkg-config libssl-dev git && \
+    curl -SfL https://github.com/solana-labs/solana/releases/download/v1.14.18/solana-release-x86_64-unknown-linux-gnu.tar.bz2 | tar jxf - -C $HOME
 
 ENV CARGO_INCREMENTAL=1
 
 ARG AUDIUS_ETH_REGISTRY_PRIVATE_KEY
 ARG TRACK_LISTEN_COUNT_PRIVATE_KEY
 ARG CLAIMABLE_TOKENS_PRIVATE_KEY
 ARG REWARD_MANAGER_PRIVATE_KEY
-
-COPY . .
-RUN --mount=type=cache,target=/usr/src/app/target \
-    --mount=type=cache,target=/root/.cache/solana \
-    --mount=type=cache,target=/usr/local/cargo/registry \
+ARG PAYMENT_ROUTER_PRIVATE_KEY
+
+COPY audius_eth_registry ./audius_eth_registry
+COPY claimable-tokens ./claimable-tokens
+COPY cli ./cli
+COPY reward-manager ./reward-manager
+COPY payment-router ./payment-router
```

Comment:
nit: Do we need the anchor programs on the bpf-builder? and do we need the other programs on the anchor-builder?

Reply:
For now, yes. But this is only because `updateKeys.sh` is written to update all programs at the same time and will throw errors if any of the files are missing. It could be split out or parameterized so that it works in each individual stage against only the programs we build in that stage.
We also run it on the last stage _again_ to update the keys in the code copied in there. And we could presumably copy the modified code out of the builder stages instead?